smart_open/s3.py

```diff
@@ -352,9 +358,9 @@ def __init__(self, bucket, key, version_id=None, buffer_size=DEFAULT_BUFFER_SIZE
             self._content_length = _get(self._object, self._version_id)['ContentLength']
         except botocore.client.ClientError as error:
             raise IOError(str(error) + "\n" +
-                          '%r does not exist in the bucket %r, version_id %r'
-                          'or is forbidden for access' % (key, bucket, self._version_id)
-                          )
+                '%r does not exist in the bucket %r, version_id %r'
```

Comment:
OK, we should _definitely_ move the exception handling to _get to avoid duplicating code like this.